### PR TITLE
ISSUE # 54: Configure member-portal to use Postgres

### PR DESCRIPTION
**Pull request by JackCurrie**
- Configure member-portal to use postgres

### CODE REVIEW

**Comment by JackCurrie** [comment]
Let me know if there's anything ya think I missed, or anything else I can do!

**Comment by JackCurrie** [comment]
https://github.com/renocollective/member-portal/issues/54

**Comment by colinloretz** [comment]
@JackCurrie check out these conflicts before we review :)

**Comment by JackCurrie** [comment]
I believe that I've got the merge conflicts all resolved. Never changed DB software for a rails app before and messed around with the config-- so definitely check to make sure that it's looking all good (I could be wrong/using bad practices 👍)

File: Gemfile [hunk 1]
```diff
@@ -4,8 +4,8 @@ source 'https://rubygems.org'
 
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
 gem 'rails', '~> 5.1.4'
-# Use sqlite3 as the database for Active Record
-gem 'sqlite3'
+# use postgres for database
+gem 'pg'
```
Comment (veronicacannon) [inline Gemfile]: Probably wise to use pessimistic versioning.

File: config/database.yml [hunk 2]
```diff
@@ -1,25 +1,23 @@
...
-  adapter: sqlite3
+  adapter: postgresql 
   pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+  encoding: unicode
   timeout: 5000
 
 development:
   <<: *default
-  database: db/development.sqlite3
+  database: member_portal_development
```
Comment (veronicacannon) [inline config/database.yml]: Might want to make the database names consistent across the environments.  Use all hyphens or all underscores.

File: config/database.yml [hunk 3]
```diff
@@ -1,25 +1,23 @@
...
   <<: *default
-  database: db/test.sqlite3
+  database: member_portal_test 
 
 production:
   <<: *default
-  database: db/production.sqlite3
+  database: member_portal_production
+  username: member_portal
+  password: <%= ENV['MEMBER_PORTAL_DATABASE_PASSWORD'] %>
```
Comment (veronicacannon) [inline config/database.yml]: Consider updating .env with this value

File: db/schema.rb [hunk 4]
```diff
@@ -1,8 +1,20 @@
-# frozen_string_literal: true
-
+# This file is auto-generated from the current state of the database. Instead
```
Comment (phlipper) [inline db/schema.rb]: Please keep the `# frozen_string_literal: true` pragma.

File: db/schema.rb [hunk 5]
```diff
@@ -13,4 +25,5 @@
     t.string "bio"
     t.string "avatar"
   end
+
```
Comment (phlipper) [inline db/schema.rb]: Please try to minimize the whitespace-only changes where possible.

**Comment by phlipper** [comment]
Please rebase this branch on top of the latest `master` to remove the merge commits.

**Comment by JackCurrie** [comment]
@phlipper thank you for the detailed feedback! Really appreciate it, its great to be able to learn from people. I'll make those changes at some point in the near future.

**Comment by JackCurrie** [comment]
Hey @phlipper !

We took a look at this and made some changes this evening. I've talked with Josh regarding the changed to the docker-compose file that you were proposing, and he said that he would look into it in another PR.

File: .env [hunk 6]
```diff
@@ -8,4 +8,6 @@
 # reference these attributes in code like this:
 # config.fog_directory  = ENV['S3_BUCKET']
 
-S3_BUCKET=bogus-key-for-example
\ No newline at end of file
+SECRET_KEY_BASE=super-fake-bogus-secret-key-base
```
Comment (phlipper) [inline .env]: The changes to `SECRET_KEY_BASE` and `S3_BUCKET` seem unrelated to this PR.

File: .gitignore [hunk 7]
```diff
@@ -25,3 +25,5 @@
 public/uploads
 
 .env.local
+
+secrets.yml
```
Comment (phlipper) [inline .gitignore]: This should not be here as it is actively being tracked by `git`.

File: config/database.yml [hunk 8]
```diff
@@ -1,25 +1,16 @@
...
-  adapter: sqlite3
+  adapter: postgresql
   pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+  encoding: unicode
   timeout: 5000
 
 development:
   <<: *default
-  database: db/development.sqlite3
+  database: <%= Rails.application.secrets["database_url"] %>
```
Comment (phlipper) [inline config/database.yml]: Per [the Rails documentation](http://guides.rubyonrails.org/configuring.html#configuring-a-database), the `DATABASE_URL` URI should use the `url` key _(vs. `database`)_ in the `config.yml`. We should also enable easy use of the URL format consistently across environments vs. just in `development`, as this is required for deployment to 12-factor environments such as Heroku.

**Comment by JackCurrie** [comment]
Thanks for the feedback, Phil! Really appreciate it. Got lots of school going on at the moment, so time to work on the project has been scarce. I'll take care of these.

File: .env (hunk 6 above)
Comment (JackCurrie) [reply .env]: So the issue with this/why we included this fake secret key here is that when "SECRET_KEY_BASE" was accessed with the "fetch" method rather than through normal hash access in ```config/secrets.yml```, it would throw an error unless we actually defined some sort of "SECRET_KEY_BASE" for the fetch to actually grab in the ```.env``` file.

Is this ok to leave in here for now? Or should we just take out the reference to this from ```config/secrets.yml``` in the first place?

File: config/database.yml (hunk 8 above)
Comment (JackCurrie) [reply config/database.yml]: cool, thanks for the link to the docs ( should have been the first place I looked :) ).

So the Fix ought to just be switching the line to ```url: <%= Rails.application.secrets["database_url"] %> ```  ?

File: .env (hunk 6 above)
Comment (phlipper) [reply .env]: @JackCurrie this has been addressed in #95. You can rebase on top of the latest `master` and then add the `DATABASE_URL` here and add the `database_url` key _(using `ENV.fetch`)_ to the `defaults` section in `secrets.yml`.

File: config/database.yml (hunk 8 above)
Comment (phlipper) [reply config/database.yml]: > So the Fix ought to just be switching the line to `url: <%= Rails.application.secrets["database_url"] %>`?

Yes, but I think Rails is now preferring Symbols to strings for the secrets:

```ruby
Rails.application.secrets[:database_url]
```
